llvm/lib/Target/AMDGPU/AMDGPUHSAMetadataStreamer.cpp

```diff
@@ -663,7 +663,10 @@ void MetadataStreamerMsgPackV5::emitHiddenKernelArgs(
     Offset += 8; // Skipped.
   }
 
-  Offset += 72; // Reserved.
+  // emit argument for hidden dynamic lds size
```

Comment:
Should skip if dynamic LDS isn't used?

Reply:
Runtime will always have the dynamic lds size passed in via hipLaunchKernelGGL(zero by default). So, it can fill this argument always. Also, there is no attribute present which I can find, which can be used to check if dynamic LDS is used in the function or not. Please let me know if we really need to skip adding this argument.

Reply:
Iterate over all the globals used by function and check that AMDGPU::isDynamicLDS(GV) is true for any global. This could be one way to check if "hidden_dynamic_lds_size" kernel arg needs to be emitted or not.

Reply:
As discussed, extended AMDGPUMachineFunction to enable a "UsesDynamicLDS" boolean flag. This is used to check if kernel uses dynamic LDS and also to decide to emit "dynamic_lds_size" hidden kernel arg. Please review.